frontend2/src/api/helpers.ts

```diff
@@ -84,3 +86,42 @@ export const prefetchNextPage = async <
     }
   }
 };
+
+const safeLoaderHelper = async <T, K>(
+  request: T,
+  factory: QueryFactory<T, K> | PaginatedQueryFactory<T, K>,
+  queryClient: QueryClient,
+): Promise<void> => {
+  try {
+    await queryClient.ensureQueryData({
+      queryKey: buildKey(factory.queryKey, request),
+      queryFn: async () => await factory.queryFn(request, queryClient, true),
+    });
+  } catch (err) {
+    if ((err as ResponseError).response.status > 500) {
+      toast.error((err as ResponseError).message);
+    }
+  }
+};
+
+/**
+ * Given a request and query identification, safely ensure that the query data exists
+ * or is fetched with no risk of throwing a Runtime Error. If the request has a server
+ * error, it will be displayed as a toast.
+ * @param request the parameters of the request
+ * @param queryKey
+ * @param queryFn
+ * @param queryClient
+ */
+export const safeLoader = <T, K>(
+  request: T,
+  factory: QueryFactory<T, K> | PaginatedQueryFactory<T, K>,
+  queryClient: QueryClient,
+): void => {
+  safeLoaderHelper(request, factory, queryClient).catch((error) => {
+    let _ = 0;
```

Comment:
Know of any better ways to trick the linter into not complaining that we're not "handling the error", even though the error handling is upstream? Maybe an eslint disable?

Reply:
we can put `void` in front of safeLoaderHelper